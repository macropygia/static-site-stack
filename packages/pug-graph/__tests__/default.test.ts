import path from 'node:path'

import fg from 'fast-glob'
import { describe, beforeAll, afterAll, test, expect } from 'vitest'

import PugGraph from '../src/index.js'

const indexPug = '__tests__/src/index.pug'
const indexPugAbs = path.resolve('__tests__/src/index.pug')
const templatePug = '__tests__/src/templates/_global.pug'

describe('Single File', () => {
  let graph: PugGraph
  beforeAll(async () => {
    graph = new PugGraph({
      baseDir: '__tests__/src',
    })
  })
  test('parse', async () => {
    await graph.parse(indexPug)
    expect(JSON.stringify(graph.getRawData())).toMatchSnapshot()
  })
  test('parse (insertOnly)', async () => {
    await graph.parse(indexPug, { insertOnly: true })
    expect(JSON.stringify(graph.getRawData())).toMatchSnapshot()
  })
  test('unlink', async () => {
    graph.unlink(indexPug)
    expect(JSON.stringify(graph.getRawData())).toMatchSnapshot()
  })
  afterAll(async () => {
    graph.exit()
  })
})

describe('Single File (Recursive)', () => {
  let graph: PugGraph
  const cwdRe = new RegExp(process.cwd(), 'g')
  beforeAll(async () => {
    graph = new PugGraph({
      baseDir: '__tests__/src',
      useAbsPath: true,
    })
  })
  test('parse (recursive, insertOnly)', async () => {
    await graph.parse(indexPugAbs, { recursive: true, insertOnly: true })
    const rawJson = JSON.stringify(graph.getRawData())
    expect(rawJson.replace(cwdRe, 'cwd')).toMatchSnapshot()
  })
  test('parse (recursive)', async () => {
    await graph.parse(indexPugAbs, { recursive: true })
    const rawJson = JSON.stringify(graph.getRawData())
    expect(rawJson.replace(cwdRe, 'cwd')).toMatchSnapshot()
  })
  afterAll(async () => {
    graph.exit()
  })
})

describe('Multiple Files', () => {
  let graph: PugGraph
  beforeAll(async () => {
    graph = new PugGraph({
      baseDir: '__tests__/src',
    })
    const files = fg.sync('__tests__/src/**/*.pug')
    await Promise.all(
      files.map((file) => graph.parse(file, { insertOnly: true })),
    )
  })
  test('getImportedFiles', async () => {
    expect(
      JSON.stringify([...graph.getImportedFiles(indexPug)]),
    ).toMatchSnapshot()
  })
  test('getImporters', async () => {
    expect(
      JSON.stringify([...graph.getImporters(templatePug)]),
    ).toMatchSnapshot()
  })
  test('getImporters (ignorePartial)', async () => {
    expect(
      JSON.stringify([...graph.getImporters(templatePug, true)]),
    ).toMatchSnapshot()
  })
  test('getRawData', async () => {
    expect(JSON.stringify(graph.getRawData())).toMatchSnapshot()
  })
  afterAll(async () => {
    graph.exit()
  })
})
