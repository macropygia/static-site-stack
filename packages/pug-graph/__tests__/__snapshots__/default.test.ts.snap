// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Multiple Files > getImportedFiles 1`] = `"["__tests__/src/templates/_category.pug","__tests__/src/templates/_global.pug","__tests__/src/templates/_base.pug","__tests__/src/templates/mixins/_index.pug","__tests__/src/templates/mixins/_mixin1.pug","__tests__/src/templates/mixins/_mixin2.pug","__tests__/src/partials/_partial1.pug","__tests__/src/foo.svg","__tests__/src/bar.html","__tests__/src/partials/_partial2.pug","__tests__/src/partials/_partial3.pug","__tests__/src/partials/_partial4.pug","__tests__/src/partials/_partial5.pug","__tests__/src/partials/_partial6.pug","__tests__/src/partials/_partial7.pug","__tests__/src/partials/_partial8.pug","__tests__/src/partials/_partial9.pug","__tests__/src/partials/_partial10.pug"]"`;

exports[`Multiple Files > getImporters (ignorePartial) 1`] = `"["__tests__/src/sub.pug","__tests__/src/index.pug"]"`;

exports[`Multiple Files > getImporters 1`] = `"["__tests__/src/sub.pug","__tests__/src/templates/_category.pug","__tests__/src/index.pug"]"`;

exports[`Multiple Files > getRawData 1`] = `"[{"path":"__tests__/src/index.pug","deps":["__tests__/src/templates/_category.pug","__tests__/src/partials/_partial1.pug","__tests__/src/foo.svg","__tests__/src/bar.html","__tests__/src/partials/_partial2.pug","__tests__/src/partials/_partial3.pug","__tests__/src/partials/_partial4.pug","__tests__/src/partials/_partial5.pug","__tests__/src/partials/_partial6.pug","__tests__/src/partials/_partial7.pug","__tests__/src/partials/_partial8.pug","__tests__/src/partials/_partial9.pug","__tests__/src/partials/_partial10.pug"]},{"path":"__tests__/src/sub.pug","deps":["__tests__/src/templates/_global.pug","__tests__/src/partials/_partial1.pug","__tests__/src/partials/_partial2.pug","__tests__/src/partials/_partial3.pug"]},{"path":"__tests__/src/partials/_partial1.pug","deps":[]},{"path":"__tests__/src/partials/_partial10.pug","deps":[]},{"path":"__tests__/src/partials/_partial2.pug","deps":[]},{"path":"__tests__/src/partials/_partial3.pug","deps":[]},{"path":"__tests__/src/partials/_partial4.pug","deps":[]},{"path":"__tests__/src/partials/_partial5.pug","deps":[]},{"path":"__tests__/src/partials/_partial6.pug","deps":[]},{"path":"__tests__/src/partials/_partial7.pug","deps":[]},{"path":"__tests__/src/partials/_partial8.pug","deps":[]},{"path":"__tests__/src/partials/_partial9.pug","deps":[]},{"path":"__tests__/src/partials/_should_ignore1.pug","deps":[]},{"path":"__tests__/src/partials/_should_ignore2.pug","deps":[]},{"path":"__tests__/src/partials/_should_ignore3.pug","deps":[]},{"path":"__tests__/src/partials/_should_ignore4.pug","deps":[]},{"path":"__tests__/src/partials/_should_ignore5.pug","deps":[]},{"path":"__tests__/src/templates/_base.pug","deps":[]},{"path":"__tests__/src/templates/_category.pug","deps":["__tests__/src/templates/_global.pug"]},{"path":"__tests__/src/templates/_global.pug","deps":["__tests__/src/templates/_base.pug","__tests__/src/templates/mixins/_index.pug"]},{"path":"__tests__/src/templates/mixins/_index.pug","deps":["__tests__/src/templates/mixins/_mixin1.pug","__tests__/src/templates/mixins/_mixin2.pug"]},{"path":"__tests__/src/templates/mixins/_mixin1.pug","deps":[]},{"path":"__tests__/src/templates/mixins/_mixin2.pug","deps":[]},{"path":"__tests__/src/templates/mixins/_should_ignore_mixin.pug","deps":[]}]"`;

exports[`Single File (Recursive) > parse (recursive) 1`] = `"[{"path":"cwd/__tests__/src/index.pug","deps":["cwd/__tests__/src/templates/_category.pug","cwd/__tests__/src/partials/_partial1.pug","cwd/__tests__/src/foo.svg","cwd/__tests__/src/bar.html","cwd/__tests__/src/partials/_partial2.pug","cwd/__tests__/src/partials/_partial3.pug","cwd/__tests__/src/partials/_partial4.pug","cwd/__tests__/src/partials/_partial5.pug","cwd/__tests__/src/partials/_partial6.pug","cwd/__tests__/src/partials/_partial7.pug","cwd/__tests__/src/partials/_partial8.pug","cwd/__tests__/src/partials/_partial9.pug","cwd/__tests__/src/partials/_partial10.pug"]},{"path":"cwd/__tests__/src/templates/_category.pug","deps":["cwd/__tests__/src/templates/_global.pug"]},{"path":"cwd/__tests__/src/partials/_partial1.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial2.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial3.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial4.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial5.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial6.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial7.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial8.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial9.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial10.pug","deps":[]},{"path":"cwd/__tests__/src/templates/_global.pug","deps":["cwd/__tests__/src/templates/_base.pug","cwd/__tests__/src/templates/mixins/_index.pug"]},{"path":"cwd/__tests__/src/templates/_base.pug","deps":[]},{"path":"cwd/__tests__/src/templates/mixins/_index.pug","deps":["cwd/__tests__/src/templates/mixins/_mixin1.pug","cwd/__tests__/src/templates/mixins/_mixin2.pug"]},{"path":"cwd/__tests__/src/templates/mixins/_mixin1.pug","deps":[]},{"path":"cwd/__tests__/src/templates/mixins/_mixin2.pug","deps":[]}]"`;

exports[`Single File (Recursive) > parse (recursive, insertOnly) 1`] = `"[{"path":"cwd/__tests__/src/index.pug","deps":["cwd/__tests__/src/templates/_category.pug","cwd/__tests__/src/partials/_partial1.pug","cwd/__tests__/src/foo.svg","cwd/__tests__/src/bar.html","cwd/__tests__/src/partials/_partial2.pug","cwd/__tests__/src/partials/_partial3.pug","cwd/__tests__/src/partials/_partial4.pug","cwd/__tests__/src/partials/_partial5.pug","cwd/__tests__/src/partials/_partial6.pug","cwd/__tests__/src/partials/_partial7.pug","cwd/__tests__/src/partials/_partial8.pug","cwd/__tests__/src/partials/_partial9.pug","cwd/__tests__/src/partials/_partial10.pug"]},{"path":"cwd/__tests__/src/templates/_category.pug","deps":["cwd/__tests__/src/templates/_global.pug"]},{"path":"cwd/__tests__/src/partials/_partial1.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial2.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial3.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial4.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial5.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial6.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial7.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial8.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial9.pug","deps":[]},{"path":"cwd/__tests__/src/partials/_partial10.pug","deps":[]},{"path":"cwd/__tests__/src/templates/_global.pug","deps":["cwd/__tests__/src/templates/_base.pug","cwd/__tests__/src/templates/mixins/_index.pug"]},{"path":"cwd/__tests__/src/templates/_base.pug","deps":[]},{"path":"cwd/__tests__/src/templates/mixins/_index.pug","deps":["cwd/__tests__/src/templates/mixins/_mixin1.pug","cwd/__tests__/src/templates/mixins/_mixin2.pug"]},{"path":"cwd/__tests__/src/templates/mixins/_mixin1.pug","deps":[]},{"path":"cwd/__tests__/src/templates/mixins/_mixin2.pug","deps":[]}]"`;

exports[`Single File > parse (insertOnly) 1`] = `"[{"path":"__tests__/src/index.pug","deps":["__tests__/src/templates/_category.pug","__tests__/src/partials/_partial1.pug","__tests__/src/foo.svg","__tests__/src/bar.html","__tests__/src/partials/_partial2.pug","__tests__/src/partials/_partial3.pug","__tests__/src/partials/_partial4.pug","__tests__/src/partials/_partial5.pug","__tests__/src/partials/_partial6.pug","__tests__/src/partials/_partial7.pug","__tests__/src/partials/_partial8.pug","__tests__/src/partials/_partial9.pug","__tests__/src/partials/_partial10.pug"]}]"`;

exports[`Single File > parse 1`] = `"[{"path":"__tests__/src/index.pug","deps":["__tests__/src/templates/_category.pug","__tests__/src/partials/_partial1.pug","__tests__/src/foo.svg","__tests__/src/bar.html","__tests__/src/partials/_partial2.pug","__tests__/src/partials/_partial3.pug","__tests__/src/partials/_partial4.pug","__tests__/src/partials/_partial5.pug","__tests__/src/partials/_partial6.pug","__tests__/src/partials/_partial7.pug","__tests__/src/partials/_partial8.pug","__tests__/src/partials/_partial9.pug","__tests__/src/partials/_partial10.pug"]}]"`;

exports[`Single File > unlink 1`] = `"[]"`;
